dtfs-central-api/src/v1/validation/utilisation-report-service/selected-fee-record-validator.ts

```diff
@@ -10,3 +11,24 @@ export const validateSelectedFeeRecordsAllHaveSamePaymentCurrency = (selectedFee
     throw new InvalidPayloadError('Selected fee records must all have the same payment currency');
   }
 };
+
+/**
+ * Validates that all selected fee records with payments have the same fee
+ * record payment group and that the fee record IDs match the fee record IDs on
+ * this existing payment group.
+ * @param feeRecordIds - The fee record IDs
+ */
+export const validateThatAllSelectedFeeRecordsWithPaymentsFormACompletePaymentGroup = async (feeRecordIds: number[]) => {
+  const selectedFeeRecords = await FeeRecordRepo.findByIdWithPaymentsAndFeeRecords(feeRecordIds);
+
+  const firstFeeRecord = selectedFeeRecords[0];
+
+  if (firstFeeRecord.status === FEE_RECORD_STATUS.DOES_NOT_MATCH) {
```

Comment:
What happens if the first fee record status is TO_DO but the second is DOES_NOT_MATCH?

Reply:
Thanks for pointing out - added a status check to ensure all statuses match. 

Covers the case where the user is manipulating the status part of the ID